website/src/docs/hotchocolate/fetching-data/resolvers.md

```diff
@@ -0,0 +1,536 @@
+            .ResolveWith<FooResolvers>(r => r.GetFoo(default, default));
+    }
+}
+```
+
+# Arguments
+
+We can access arguments we defined for our resolver like regular arguments of a function.
+
+There are also specific arguments that will be automatically populated by Hot Chocolate when the resolver is executed. These include [Dependency injection services](#injecting-services), [DataLoaders](/docs/hotchocolate/fetching-data/dataloader), state, or even context like a [_parent_](#accessing-parent-values) value.
+
+[Learn more about arguments](/docs/hotchocolate/defining-a-schema/arguments)
+
+# Injecting Services
+
+Resolvers integrate nicely with `Microsoft.Extensions.DependecyInjection`.
+We can access all registered services in our resolvers.
+
+Let's assume we have created a `UserService` and registered it as a service.
+
+```csharp
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSingleton<UserService>()
+
+        services
+            .AddGraphQLServer()
+            .AddQueryType<Query>();
+    }
+}
+```
+
+We can then access the `UserService` in our resolvers like the following.
+
+<ExampleTabs>
+<ExampleTabs.Annotation>
+
+```csharp
+public class Query
+{
+    public List<User> GetUsers([Service] UserService userService)
+        => userService.GetUsers();
+}
+```
+
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
+
+```csharp
+public class Query
+{
+    public List<User> GetUsers([Service] UserService userService)
+        => userService.GetUsers();
+}
+
+public class QueryType: ObjectType<Query>
+{
+    protected override void Configure(IObjectTypeDescriptor<Query> descriptor)
+    {
+        descriptor
+            .Field(f => f.Foo(default))
+            .Type<ListType<UserType>>();
+    }
+}
+```
+
+When using the `Resolve` method, we can access services through the `IResolverContext`.
+
+```csharp
+descriptor
+    .Field("foo")
+    .Resolve(context =>
+    {
+        var userService = context.Service<UserService>();
+
+        return userService.GetUsers();
+    });
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+```csharp
+public class Query
+{
+    public List<User> GetUsers([Service] UserService userService)
+        => userService.GetUsers();
+}
+```
+
+When using `AddResolver()`, we can access services through the `IResolverContext`.
+
+```csharp
+services
+    .AddGraphQLServer()
+    .AddDocumentFromString(@"
+        type Query {
+          users: [User!]!
+        }
+    ")
+    .AddResolver("Query", "users", (context) =>
+    {
+        var userService = context.Service<UserService>();
+
+        return userService.GetUsers();
+    });
+```
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+Hot Chocolate will correctly instantiate the service, depending on its lifetime. A scoped service for example is only instantiated once per request and this instance is injected into all resolvers used during the request.
```

Comment:
Damn, I somehow didn't submit this comment a while back.

Basically just want to:
1. use inject instead of instantiated (as if it was singleton it wouldn't necessaily always be istantianted).
2. make it clear that scoped isn't always necessarily request level.